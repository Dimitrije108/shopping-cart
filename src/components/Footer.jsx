export default function Footer() {
  return (
		<div>
			This here is a Footer placeholder for now!
		</div>
	)
};