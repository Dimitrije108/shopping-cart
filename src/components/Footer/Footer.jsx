import { Link } from "react-router-dom";
import styles from "./Footer.module.css";
import layout from "../../layout.module.css";
// assets
import locationIcon from "/src/assets/icons/location.svg";
import emailIcon from "/src/assets/icons/email.svg";
// import websiteIcon from "/src/assets/icons/website.svg";
import CEC from "/src/assets/icons/Corellian_Engineering_Corporation.svg";
import federationLogo from "/src/assets/icons/Corellian_Federation_logo.webp";
import kuatDriveYardsLogo from "/src/assets/icons/Kuat-Drive-Yards.svg";
import fondorShipyardsLogo from "/src/assets/icons/Fondor-Shipyards.jpg";
import githubIcon from "/src/assets/icons/github-mark-white.svg";

export default function Footer() {
  return (
		<div className={styles.footerCont}>
			<div className={`${styles.mainFooterWrapper} ${layout.layoutWrapper}`}>
				<div className={styles.contactCont}>
					<div className={styles.locationCont}>
						<div className={styles.iconCont}>
							<img 
								src={locationIcon}
								alt="location icon" 
								width={30}
								title="Location"
							/>
						</div>
						<div className={styles.location}>
							<p className={styles.address}>Orbital Assembly Facility 7</p>
							<p>Corellia, Corellian System</p>
							<p>Corellia Sector, Core Worlds</p>
						</div>
					</div>
					<div className={styles.emailCont}>
						<div className={styles.iconCont}>
							<img 
								src={emailIcon} 
								alt="email icon" 
								width={30}
								title="Email"
							/>
						</div>
						<p>sw_shipyard@corellia.cec</p>
					</div>
				</div>
				<div className={styles.companyCont}>
					<button>About Us</button>
					<button>Our Services</button>
					<button>Privacy Policy</button>
					<button>Contact</button>
				</div>
				<div className={styles.partnersCont}>
					<div className={styles.iconWrapper}>
						<div className={styles.partnersTopRow}>
							<Link 
								to={"https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation"}
								className={styles.linkEl}
								data-testid="cecLink"
							>
								<img 
									src={CEC}
									alt="Corellian Engineering Corporation icon" 
									width={55}
									title="Corellian Engineering Corporation"
								/>
							</Link>
							<Link 
								to={"https://starwars.fandom.com/wiki/Confederation"}
								className={styles.linkEl}
								data-testid="cfLink"
							>
								<img 
									src={federationLogo}
									alt="Corellian Federation icon" 
									width={55}
									title="Corellian Federation"
								/>
							</Link>
						</div>
						<div className={styles.partnersBottomRow}>
							<Link 
								to={"https://starwars.fandom.com/wiki/Kuat_Drive_Yards"}
								className={styles.linkEl}
								data-testid="kdyLink"
							>
								<img 
									src={kuatDriveYardsLogo}
									alt="Kuat Drive Yards icon" 
									width={55}
									title="Kuat Drive Yards"
								/>
							</Link>
							<Link 
								to={"https://starwars.fandom.com/wiki/Fondor_Shipyards"}
								className={styles.linkEl}
								data-testid="fondorLink"
							>
								<img 
									src={fondorShipyardsLogo}
									alt="Fondor Shipyards icon" 
									width={55}
									title="Fondor Shipyards"
								/>
							</Link>
						</div>
					</div>
				</div>
			</div>
			<div className={styles.copyrightCont}>
				<div className={styles.copyrightText}>
					<p>Copyright Â© 2025 Star Wars Shipyard limited. All rights Reserved.</p>
					<p className={styles.disclaimer}>*This is a placeholder. It is a fan project made with no commercial desire in mind.</p>
				</div>
				<Link 
					to="https://github.com/Dimitrije108/shopping-cart"
					className={styles.githubLink}
					data-testid="githubLink"
				>
					<img 
						src={githubIcon}
						alt="github icon" 
						title="GitHub"
					/>
				</Link>
			</div>
		</div>
	)
};
