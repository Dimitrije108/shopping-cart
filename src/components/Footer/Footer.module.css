.footerCont {
	padding: 2rem 0 0.9rem 0;

	background-color: black;
	border-top: 1px solid rgb(104, 104, 104);
}

.mainFooterWrapper {
	margin: 0 auto;
	padding: 0 1rem;

	display: grid;
	grid-template-columns: 1fr 1fr 1fr;
	grid-template-areas: 
			"contact company partners";
}

.contactCont {
	grid-area: contact;
}

.locationCont {
	display: flex;
}

.location {
	font-size: 0.9rem;
}

.address {
	font-family: 'DINNext bold';
	font-size: 1rem;
}

.emailCont,
.webpageCont {
	margin-top: 0.5rem;
	display: flex;
	align-items: flex-end;
}

.iconCont {
	margin-top: 4px;
	margin-right: 12px;

	display: flex;
	justify-content: center;
	align-items: flex-start;
}

.emailCont > p {
	display: flex;
	align-items: center;
}

.webpageCont > a {
	color: rgb(68, 102, 255);
	transition: 0.3s color ease;
}

.webpageCont > a:hover {
	color: rgb(37, 77, 255);
}

.companyCont {
	font-family: Trench;

	grid-area: company;
	justify-self: center;

	display: flex;
	flex-direction: column;
}

.companyCont > button {
	margin: 0;
	padding: 8px 0;

	font-size: 1.5rem;
	text-align: start;
	border: none;
}

.companyCont > button:hover {
	color: white;
}

.partnersCont {
	grid-area: partners;
	justify-self: end;
	border: 1px solid rgb(104, 104, 104);
}

.iconWrapper {
	margin: 4px;
	padding: 6px 10px;

	background-color: white;
}

.partnersTopRow {
	margin-top: 6px;
}

.partnersTopRow > a,
.partnersBottomRow > a {
	margin: 6px 10px 4px 10px;
	cursor: pointer;
}

.copyrightCont {
	font-size: 0.9rem;
	text-align: center;

	display: flex;
	flex-direction: column;
}

.copyrightText {
	margin: 1.5rem 0 0.8rem 0;
	padding: 1rem;
	background-color: rgb(20, 20, 20);
}

.disclaimer {
	font-size: 0.8rem;
	color: rgb(148, 148, 148);
}

.githubLink {
	align-self: center;
}

.githubLink > img {
	width: 35px;
}

.linkEl {
	line-height: 0;
}

@media (max-width: 750px) {
	.mainFooterWrapper {
		grid-template-columns: 1fr 1fr;
		grid-template-rows: 1fr 1fr;
		gap: 1rem;

		grid-template-areas: 
			"contact company"
			"partners partners";
	}

	.contactCont {
		grid-area: contact;
		justify-self: center;
	}

	.companyCont {
		grid-area: company;
		justify-self: center;
	}

	.partnersCont {
		grid-area: partners;
		justify-self: center;
		align-self: center;
	}

	.iconWrapper {
		display: flex;
	}
}

@media (max-width: 450px) {
	.mainFooterWrapper {
		grid-template-columns: 1fr;
		grid-template-rows: auto auto auto;
		gap: 2rem;

		grid-template-areas: 
			"company"
			"contact"
			"partners";
	}

	.contactCont {
		grid-area: contact;
		justify-self: center;
	}

	.companyCont {
		grid-area: company;
		justify-self: center;
	}

	.partnersCont {
		grid-area: partners;
		justify-self: center;
		align-self: center;
	}

	.iconWrapper {
		display: block;
	}

	.copyrightText {
		font-size: 0.8rem;
	}
	
	.disclaimer {
		font-size: 0.7rem;
	}

	.githubLink > img {
		width: 30px;
	}
}
