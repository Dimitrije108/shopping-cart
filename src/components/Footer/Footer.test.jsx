import { describe, it, expect } from "vitest";
import { render, screen  } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Footer from "./Footer";

describe("Footer component", () => {
	it("Footer component snapshot", () => {
		const footer = render(
			<MemoryRouter>
				<Footer />
			</MemoryRouter>
		)

		expect(footer).toMatchSnapshot();
	});

	describe("Footer component renders correctly", () => {
		it("Contact section renders correctly", () => {
			render(
				<MemoryRouter>
					<Footer />
				</MemoryRouter>
			)
			
			const locationIcon = screen.getByTitle("Location");
			const locationText = screen.getByText("Orbital Assembly Facility 7");
			const emailIcon = screen.getByTitle("Email");
			const email = screen.getByText("sw_shipyard@corellia.cec");

			expect(locationIcon).toBeInTheDocument();
			expect(locationIcon).toHaveAttribute("src", "/src/assets/location.svg");
			expect(locationText).toBeInTheDocument();
			expect(emailIcon).toBeInTheDocument();
			expect(emailIcon).toHaveAttribute("src", "/src/assets/email.svg");
			expect(email).toBeInTheDocument();
		});

		it("Company section renders correctly", () => {
			render(
				<MemoryRouter>
					<Footer />
				</MemoryRouter>
			)
			
			const aboutUs = screen.getByRole("button", { name: "About Us" });
			const ourServices = screen.getByRole("button", { name: "Our Services" });
			const privacyPolicy = screen.getByRole("button", { name: "Privacy Policy" });
			const contact = screen.getByRole("button", { name: "Contact" });

			expect(aboutUs).toBeInTheDocument();
			expect(ourServices).toBeInTheDocument();
			expect(privacyPolicy).toBeInTheDocument();
			expect(contact).toBeInTheDocument();
		});

		it("Partners section renders correctly", () => {
			render(
				<MemoryRouter>
					<Footer />
				</MemoryRouter>
			)
			
			const cecLink = screen.getByTestId("cecLink");
			const cecLogo = screen.getByTitle("Corellian Engineering Corporation");
			const cfLink = screen.getByTestId("cfLink");
			const cfLogo = screen.getByTitle("Corellian Federation");
			const kdyLink = screen.getByTestId("kdyLink");
			const kdyLogo = screen.getByTitle("Kuat Drive Yards");
			const fondorLink = screen.getByTestId("fondorLink");
			const fondorLogo = screen.getByTitle("Fondor Shipyards");

			expect(cecLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation");
			expect(cecLogo).toHaveAttribute("src", "/src/assets/Corellian_Engineering_Corporation.svg");
			expect(cfLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Confederation");
			expect(cfLogo).toHaveAttribute("src", "/src/assets/Corellian_Federation_logo.webp");
			expect(kdyLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Kuat_Drive_Yards");
			expect(kdyLogo).toHaveAttribute("src", "/src/assets/Kuat-Drive-Yards.svg");
			expect(fondorLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Fondor_Shipyards");
			expect(fondorLogo).toHaveAttribute("src", "/src/assets/Fondor-Shipyards.jpg");
		});

		it("Copyright section renders correctly", () => {
			render(
				<MemoryRouter>
					<Footer />
				</MemoryRouter>
			)
			
			const copyrightText = screen.getByText("Copyright Â© 2025 Star Wars Shipyard limited. All rights Reserved.");
			const disclaimerText = screen.getByText("*This is a placeholder. It is a fan project made with no commercial desire in mind.");
			const githubLink = screen.getByTestId("githubLink");
			const githubIcon = screen.getByTitle("GitHub");

			expect(copyrightText).toBeInTheDocument();
			expect(disclaimerText).toBeInTheDocument();
			expect(githubLink).toHaveAttribute("href", "https://github.com/Dimitrije108/shopping-cart");
			expect(githubIcon).toHaveAttribute("src", "/src/assets/github-mark-white.svg");
		});
	});
});
