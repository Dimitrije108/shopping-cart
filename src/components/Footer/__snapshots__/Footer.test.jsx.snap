// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > Footer component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_footerCont_eb428f"
      >
        <div
          class="_mainFooterWrapper_eb428f _layoutWrapper_aadc39"
        >
          <div
            class="_contactCont_eb428f"
          >
            <div
              class="_locationCont_eb428f"
            >
              <div
                class="_iconCont_eb428f"
              >
                <img
                  alt="location icon"
                  src="/src/assets/icons/location.svg"
                  title="Location"
                  width="30"
                />
              </div>
              <div
                class="_location_eb428f"
              >
                <p
                  class="_address_eb428f"
                >
                  Orbital Assembly Facility 7
                </p>
                <p>
                  Corellia, Corellian System
                </p>
                <p>
                  Corellia Sector, Core Worlds
                </p>
              </div>
            </div>
            <div
              class="_emailCont_eb428f"
            >
              <div
                class="_iconCont_eb428f"
              >
                <img
                  alt="email icon"
                  src="/src/assets/icons/email.svg"
                  title="Email"
                  width="30"
                />
              </div>
              <p>
                sw_shipyard@corellia.cec
              </p>
            </div>
            <div
              class="_webpageCont_eb428f"
            >
              <div
                class="_iconCont_eb428f"
              >
                <img
                  alt="webpage icon"
                  src="/src/assets/icons/website.svg"
                  title="Webpage"
                  width="30"
                />
              </div>
              <a
                data-testid="webpageLink"
                href="https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation"
              >
                Corellian Engineering Corporation
              </a>
            </div>
          </div>
          <div
            class="_companyCont_eb428f"
          >
            <button>
              <a
                data-discover="true"
                href="/#aboutUsSection"
              >
                About Us
              </a>
            </button>
            <button>
              Our Services
            </button>
            <button>
              Privacy Policy
            </button>
            <button>
              <a
                data-discover="true"
                href="/contact"
              >
                Contact
              </a>
            </button>
          </div>
          <div
            class="_partnersCont_eb428f"
          >
            <div
              class="_iconWrapper_eb428f"
            >
              <div
                class="_partnersTopRow_eb428f"
              >
                <a
                  class="_linkEl_eb428f"
                  data-testid="cecLink"
                  href="https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation"
                >
                  <img
                    alt="Corellian Engineering Corporation icon"
                    src="/src/assets/icons/Corellian_Engineering_Corporation.svg"
                    title="Corellian Engineering Corporation"
                    width="55"
                  />
                </a>
                <a
                  class="_linkEl_eb428f"
                  data-testid="cfLink"
                  href="https://starwars.fandom.com/wiki/Confederation"
                >
                  <img
                    alt="Corellian Federation icon"
                    src="/src/assets/icons/Corellian_Federation_logo.webp"
                    title="Corellian Federation"
                    width="55"
                  />
                </a>
              </div>
              <div
                class="_partnersBottomRow_eb428f"
              >
                <a
                  class="_linkEl_eb428f"
                  data-testid="kdyLink"
                  href="https://starwars.fandom.com/wiki/Kuat_Drive_Yards"
                >
                  <img
                    alt="Kuat Drive Yards icon"
                    src="/src/assets/icons/Kuat-Drive-Yards.svg"
                    title="Kuat Drive Yards"
                    width="55"
                  />
                </a>
                <a
                  class="_linkEl_eb428f"
                  data-testid="fondorLink"
                  href="https://starwars.fandom.com/wiki/Fondor_Shipyards"
                >
                  <img
                    alt="Fondor Shipyards icon"
                    src="/src/assets/icons/Fondor-Shipyards.jpg"
                    title="Fondor Shipyards"
                    width="55"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="_copyrightCont_eb428f"
        >
          <div
            class="_copyrightText_eb428f"
          >
            <p>
              Copyright Â© 2025 Star Wars Shipyard limited. All rights Reserved.
            </p>
            <p
              class="_disclaimer_eb428f"
            >
              *This is a placeholder. It is a fan project made with no commercial desire in mind.
            </p>
          </div>
          <a
            class="_githubLink_eb428f"
            data-testid="githubLink"
            href="https://github.com/Dimitrije108/shopping-cart"
          >
            <img
              alt="github icon"
              src="/src/assets/icons/github-mark-white.svg"
              title="GitHub"
            />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_footerCont_eb428f"
    >
      <div
        class="_mainFooterWrapper_eb428f _layoutWrapper_aadc39"
      >
        <div
          class="_contactCont_eb428f"
        >
          <div
            class="_locationCont_eb428f"
          >
            <div
              class="_iconCont_eb428f"
            >
              <img
                alt="location icon"
                src="/src/assets/icons/location.svg"
                title="Location"
                width="30"
              />
            </div>
            <div
              class="_location_eb428f"
            >
              <p
                class="_address_eb428f"
              >
                Orbital Assembly Facility 7
              </p>
              <p>
                Corellia, Corellian System
              </p>
              <p>
                Corellia Sector, Core Worlds
              </p>
            </div>
          </div>
          <div
            class="_emailCont_eb428f"
          >
            <div
              class="_iconCont_eb428f"
            >
              <img
                alt="email icon"
                src="/src/assets/icons/email.svg"
                title="Email"
                width="30"
              />
            </div>
            <p>
              sw_shipyard@corellia.cec
            </p>
          </div>
          <div
            class="_webpageCont_eb428f"
          >
            <div
              class="_iconCont_eb428f"
            >
              <img
                alt="webpage icon"
                src="/src/assets/icons/website.svg"
                title="Webpage"
                width="30"
              />
            </div>
            <a
              data-testid="webpageLink"
              href="https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation"
            >
              Corellian Engineering Corporation
            </a>
          </div>
        </div>
        <div
          class="_companyCont_eb428f"
        >
          <button>
            <a
              data-discover="true"
              href="/#aboutUsSection"
            >
              About Us
            </a>
          </button>
          <button>
            Our Services
          </button>
          <button>
            Privacy Policy
          </button>
          <button>
            <a
              data-discover="true"
              href="/contact"
            >
              Contact
            </a>
          </button>
        </div>
        <div
          class="_partnersCont_eb428f"
        >
          <div
            class="_iconWrapper_eb428f"
          >
            <div
              class="_partnersTopRow_eb428f"
            >
              <a
                class="_linkEl_eb428f"
                data-testid="cecLink"
                href="https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation"
              >
                <img
                  alt="Corellian Engineering Corporation icon"
                  src="/src/assets/icons/Corellian_Engineering_Corporation.svg"
                  title="Corellian Engineering Corporation"
                  width="55"
                />
              </a>
              <a
                class="_linkEl_eb428f"
                data-testid="cfLink"
                href="https://starwars.fandom.com/wiki/Confederation"
              >
                <img
                  alt="Corellian Federation icon"
                  src="/src/assets/icons/Corellian_Federation_logo.webp"
                  title="Corellian Federation"
                  width="55"
                />
              </a>
            </div>
            <div
              class="_partnersBottomRow_eb428f"
            >
              <a
                class="_linkEl_eb428f"
                data-testid="kdyLink"
                href="https://starwars.fandom.com/wiki/Kuat_Drive_Yards"
              >
                <img
                  alt="Kuat Drive Yards icon"
                  src="/src/assets/icons/Kuat-Drive-Yards.svg"
                  title="Kuat Drive Yards"
                  width="55"
                />
              </a>
              <a
                class="_linkEl_eb428f"
                data-testid="fondorLink"
                href="https://starwars.fandom.com/wiki/Fondor_Shipyards"
              >
                <img
                  alt="Fondor Shipyards icon"
                  src="/src/assets/icons/Fondor-Shipyards.jpg"
                  title="Fondor Shipyards"
                  width="55"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="_copyrightCont_eb428f"
      >
        <div
          class="_copyrightText_eb428f"
        >
          <p>
            Copyright Â© 2025 Star Wars Shipyard limited. All rights Reserved.
          </p>
          <p
            class="_disclaimer_eb428f"
          >
            *This is a placeholder. It is a fan project made with no commercial desire in mind.
          </p>
        </div>
        <a
          class="_githubLink_eb428f"
          data-testid="githubLink"
          href="https://github.com/Dimitrije108/shopping-cart"
        >
          <img
            alt="github icon"
            src="/src/assets/icons/github-mark-white.svg"
            title="GitHub"
          />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
